sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -73,7 +75,7 @@ public static BatchSpanProcessorBuilder builder(SpanExporter spanExporter) {
             scheduleDelayNanos,
             maxExportBatchSize,
             exporterTimeoutNanos,
-            new ArrayBlockingQueue<>(maxQueueSize));
+            JcTools.newMpscArrayQueue(maxQueueSize));
```

Comment:
Note that MpscArrayQueue rounds the queue size to power to 2 for various perf reasons. In my opinion it is better to enforce this so users know what the actual memory that is getting allocated.

Reply:
Do you mean falling back to `ArrayBlockingQueue` if size isn't power of 2? I don't think we can require this for the BSP setting instead since it's too tricky to use.

Reply:
I meant enforcing the maxQueueSize to be a power of 2.

Reply:
That we can't do we don't want to lose usability (adding restrictions that can only be conveyed through documentation or error messages) here. Would like to hear more thoughts on whether we should fallback if it's not power-of-2

Reply:
Falling back is not great really since it is not an efficient solution. How about calling it out in the documentation that queue size is rounded to the next power of 2?

Reply:
I added a note to the builder that some more memory may be allocated, without going too much into implementation detail.